ui/desktop/src/recipe/recipeStorage.ts

```diff
@@ -183,19 +181,34 @@ export async function listSavedRecipes(includeArchived: boolean = false): Promis
     const globalFiles = await window.electron.listFiles(globalDir, 'yaml');
     const localFiles = await window.electron.listFiles(localDir, 'yaml');
 
-    // Process global recipes
-    for (const file of globalFiles) {
+    // Process global recipes in parallel
+    const globalRecipePromises = globalFiles.map(async (file) => {
+      const recipeName = file.replace(/\.yaml$/, '');
+      return await loadRecipeFromFile(recipeName, true);
+    });
+
+    // Process local recipes in parallel
+    const localRecipePromises = localFiles.map(async (file) => {
       const recipeName = file.replace(/\.yaml$/, '');
-      const recipe = await loadRecipeFromFile(recipeName, true);
+      return await loadRecipeFromFile(recipeName, false);
+    });
+
+    // Wait for all recipes to load in parallel
+    const [globalRecipes, localRecipes] = await Promise.all([
+      Promise.all(globalRecipePromises),
+      Promise.all(localRecipePromises),
+    ]);
+
+    // Filter out null results and apply archived filter
+    const recipes: SavedRecipe[] = [];
+
+    for (const recipe of globalRecipes) {
```

Comment:
should we take this as a reason to remove the local recipes? it doesn't make sense to me

Reply:
My thinking, for local recipes was that it follows how memory and .goosehints work. Both have global/local versions which can be used by Goose.  Devs can decide whether they want to add a .goosehints and memory files at the repository level, which creates a more consistent experience/development environment for Goose users. 

I envision being able to clone a repository, open it in Goose, and find a list established recipes that demonstrate how to interact or get the most use out of a project.

Reply:
I think that uses GOOSE_RECIPE_PATH_ENV_VAR maybe? 

I think we talked about this in a different context about how this should be done through the server and not have the client poke around here; I wrote up this doc:

https://docs.google.com/document/d/1vjQn2MtH2XLC9o4OxPkremu2HPdaxwtbbKPzBJoFxpo/edit?tab=t.0#heading=h.ghh7fchtg0dw

on how I would like to see things go. local recipes are not a bad idea per se, but I think we should unify the CLI and the desktop first even if that means dropping the local recipes for now